### PR TITLE
Adding Relayer initial logic for connectivity tests

### PR DESCRIPTION
**Pull request by dylanbronson**
Adding initial logic for Relayer. This is just baseline logic. This module will be added to later once we finalize the design of the connectivity tests.

The Relayer simply receives a message and relays it onward. It is very similar to TemperatureFilter with the temperature logic removed.

### CODE REVIEW

File: edge-modules/Relayer/Program.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    class Program
+    {
+        static readonly ILogger Logger = ModuleUtil.CreateLogger("Relayer");
+
+        static async Task Main(string[] args)
+        {
+            Logger.LogInformation($"Starting Relayer with the following settings: \r\n{Settings.Current}");
+            try
+            {
+                // Create moduleClient
```
Comment (and-rewsmith) [inline edge-modules/Relayer/Program.cs]: Recommend removing

File: edge-modules/Relayer/Settings.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                    configuration.GetValue("transportType", TransportType.Amqp_Tcp_Only),
+                    configuration.GetValue("outputName", "output1"));
+            });
+
+        Settings(
+            TransportType transportType,
+            string outputName)
+        {
+            this.TransportType = transportType;
+            this.OutputName = outputName;
```
Comment (and-rewsmith) [inline edge-modules/Relayer/Settings.cs]: Recommend using Preconditions.Check statements to validate arguments not null. For an example, look at ModuleRestarter.

File: Microsoft.Azure.Devices.Edge.sln [hunk 3]
```diff
@@ -189,7 +189,9 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Azure.Devices.Edg
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Azure.Devices.Edge.Test", "test\Microsoft.Azure.Devices.Edge.Test\Microsoft.Azure.Devices.Edge.Test.csproj", "{08986FED-9283-47F1-B938-86D9C6E753E1}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Microsoft.Azure.Devices.Edge.Agent.Kubernetes.IntegrationTest", "edge-agent\test\Microsoft.Azure.Devices.Edge.Agent.Kubernetes.IntegrationTest\Microsoft.Azure.Devices.Edge.Agent.Kubernetes.IntegrationTest.csproj", "{79540E35-45EA-4568-9AB7-C1CC0BFF73EE}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Azure.Devices.Edge.Agent.Kubernetes.IntegrationTest", "edge-agent\test\Microsoft.Azure.Devices.Edge.Agent.Kubernetes.IntegrationTest\Microsoft.Azure.Devices.Edge.Agent.Kubernetes.IntegrationTest.csproj", "{79540E35-45EA-4568-9AB7-C1CC0BFF73EE}"
```
Comment (and-rewsmith) [inline Microsoft.Azure.Devices.Edge.sln]: I'm guessing this was accidental?

**Comment by philipktlin** [comment]
LGTM

File: edge-modules/Relayer/Program.cs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+                    throw new InvalidOperationException("UserContext doesn't contain expected value");
+                }
+
+                await moduleClient.SendEventAsync(Settings.Current.OutputName, message);
+                Logger.LogInformation("Successfully sent a message");
+            }
+            catch (Exception ex)
+            {
+                Logger.LogError($"Error in ProcessAndSendMessageAsync: {ex}");
+            }
```
[the remark this answers is not in the document]
Comment (philipktlin) [reply edge-modules/Relayer/Program.cs]: I think 1st version is just counting message sent, received and failed to send; no retry is needed.  If retry, do we retry until it succeeds?

**Pull request merged**